core/shared/src/main/scala/zio/FiberId.scala

```diff
@@ -45,17 +44,23 @@ sealed trait FiberId extends Serializable { self =>
       case Composite(l, r)   => l.ids ++ r.ids
     }
 
-  final def isNone: Boolean = toSet.forall(_.isNone)
+  final def isNone: Boolean =
+    self match {
+      case None                     => true
+      case FiberId.Runtime(_, _, _) => false
+      case Composite(_, _)          => false
```

Comment:
Unfortunately due to what @ghostdogpr says, you can't necessarily say this is false because a user could construct `Composite(None, None)` themselves, and it's not easy to get around that due to the requirement for backward compatibility.

Btw converting to a set is very slow. I would just recurse directly, in order to avoid any (heap) allocations. That makes it stack unsafe for large fiber ids. If you wanted to deal with that, you could catch StackOverFlow exception and fallback with a set.

Reply:
> I would just recurse directly, in order to avoid any (heap) allocations. That makes it stack unsafe for large fiber ids. If you wanted to deal with that, you could catch StackOverFlow exception and fallback with a set.

Yes, I was worried about stack safety that's why I suggested using the `toSet` in my solution 2. here: https://github.com/zio/zio/pull/8746#discussion_r1565439528
I didn't know about the possibility of catching `StackOverFlow`. I'll implement that. Thanks!

Reply:
Implemented